modules/mockk/src/jvmTest/kotlin/io/mockk/impl/recording/JvmSignatureValueGeneratorTest.kt

```diff
@@ -0,0 +1,82 @@
+package io.mockk.impl.recording
+
+import io.mockk.impl.instantiation.AbstractInstantiator
+import io.mockk.impl.instantiation.AnyValueGenerator
+import io.mockk.impl.instantiation.CommonInstanceFactoryRegistry
+import kotlin.reflect.KClass
+import kotlin.test.Test
+import kotlin.test.assertTrue
+import kotlin.test.assertNotNull
+import kotlin.test.fail
+import kotlin.time.Duration
+import java.util.Random
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+
+class JvmSignatureValueGeneratorTest {
+
+    private val generator = JvmSignatureValueGenerator(Random(42))
+
+    private val mockInstantiator = object : AbstractInstantiator(CommonInstanceFactoryRegistry()) {
+        override fun <T : Any> instantiate(cls: KClass<T>): T = fail("Should not instantiate")
+    }
+
+    private val mockAnyValueGenerator = object : AnyValueGenerator() {
+        override fun anyValue(
+            cls: KClass<*>,
+            isNullable: Boolean,
+            orInstantiateVia: () -> Any?
+        ): Any? = fail("Should not generate any value")
+    }
+
+    @Test
+    fun `generated Long values avoid denormalized range`() {
+        // Calculate denormalized range boundaries (same as implementation)
+        val nanosInMillis = 1_000_000L
+        val maxNanos = Long.MAX_VALUE / 2 / nanosInMillis * nanosInMillis - 1
+        val maxNanosInMillis = maxNanos / nanosInMillis
+
+        repeat(10) {
```

Comment:
I see what you're doing here and it makes sense to me.

I have a few possible improvements I can think of:
- I'd run each of these more than just 10 times, as the error in #1401 happens less frequently than 1 in 10 times.
- I'd extract the number of iterations in a constant and reuse it across all tests
- Can we run these iterations in parallel to speed up the test execution?